data/tokenizer.py

```diff
@@ -120,3 +128,75 @@ def untokenize(self, tokens: list[str]) -> pd.DataFrame:
         df = pd.DataFrame(velocities, columns=["velocity"])
 
         return df
+
+
+class DstartEncoder(MidiEncoder):
+    def __init__(self, bins: int = 200):
+        super().__init__()
+        self.specials = ["<s>", "</s>", "<blank>"]
+        self.bins = bins
+        # Make a copy of special tokens ...
+        self.vocab = list(self.specials)
+        # ... and add velocity tokens
+        self._build_vocab()
+        self._bin_edges = self._load_bin_edges()
+        self.bin_to_dstart = []
+        self._build_dstart_decoder()
+        self.token_to_id = {token: it for it, token in enumerate(self.vocab)}
+
+    def _load_bin_edges(self):
+        artifacts_path = to_absolute_path("artifacts/bin_edges.yaml")
+        with open(artifacts_path, "r") as f:
+            bin_edges = yaml.safe_load(f)
+
+        dstart_bin_edges = bin_edges["dstart"][self.bins]
+        return dstart_bin_edges
+
+    def _build_vocab(self):
+        self.vocab += [str(possible_bin) for possible_bin in range(self.bins)]
+
+    def _build_dstart_decoder(self):
+        self.bin_to_dstart = []
+        for it in range(1, len(self._bin_edges)):
+            dstart = (self._bin_edges[it - 1] + self._bin_edges[it]) / 2
+            self.bin_to_dstart.append(dstart)
+
+        last_dstart = 2 * self._bin_edges[-1]
+        self.bin_to_dstart.append(last_dstart)
+
+    @property
+    def vocab_size(self) -> int:
+        return len(self.vocab)
+
+    def unquantized_start(self, dstart_bins: pd.Series) -> pd.Series:
```

Comment:
```suggestion
    def unquantized_start(self, dstart_bins: np.array) -> np.array:
```

I think we don't need `pd.Series` here, numpy should make it more accessible